### PR TITLE
Feature/table graph time axis

### PR DESCRIPTION
**Pull request by ebb-tide**
- [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [ ] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #

### The problem
Flipping time axis. 
### The Solution
Pre-flip time axis at time of request

### CODE REVIEW

File: ui/src/dashboards/components/GraphOptionsTimeAxis.js [hunk 1]
```diff
@@ -1,29 +1,30 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-const VERTICAL = 'VERTICAL'
-const HORIZONTAL = 'HORIZONTAL'
-const GraphOptionsTimeAxis = ({TimeAxis, onToggleTimeAxis}) =>
+const GraphOptionsTimeAxis = ({verticalTimeAxis, onToggleVerticalTimeAxis}) =>
```
Comment (ischolten) [inline ui/src/dashboards/components/GraphOptionsTimeAxis.js]: this component could be converted to typescript

File: ui/src/dashboards/components/TableOptions.tsx [hunk 2]
```diff
@@ -66,12 +66,11 @@ export class TableOptions extends PureComponent<Props, {}> {
       queryConfigs.map(({measurement, fields}) => {
         return fields.map(({alias}) => {
           const internalName = `${measurement}.${alias}`
-          const existing = this.columnNames.find(
-            c => c.internalName === internalName
-          )
+          const existing = this.columnNames.find(c => c.internalName === internalName)
```
Comment (ischolten) [inline ui/src/dashboards/components/TableOptions.tsx]: this looks like a prettier change.

File: ui/src/dashboards/components/TableOptions.tsx [hunk 3]
```diff
@@ -101,48 +92,37 @@ export class TableOptions extends PureComponent<Props, {}> {
...
 
   handleToggleTextWrapping = () => {}
 
   handleColumnRename = column => {
     const {handleUpdateTableOptions, tableOptions} = this.props
     const {columnNames} = tableOptions
-    const updatedColumns = columnNames.map(
-      op => (op.internalName === column.internalName ? column : op)
-    )
+    const updatedColumns = columnNames.map(op => (op.internalName === column.internalName ? column : op))
```
Comment (ischolten) [inline ui/src/dashboards/components/TableOptions.tsx]: prettier change

File: ui/src/dashboards/components/TableOptions.tsx [hunk 4]
```diff
@@ -101,48 +92,37 @@ export class TableOptions extends PureComponent<Props, {}> {
...
 
   render() {
-    const {
-      tableOptions: {timeFormat, columnNames: columns},
-      onResetFocus,
-      tableOptions,
-    } = this.props
-
-    const TimeAxis = 'vertical'
+    const {tableOptions: {timeFormat, columnNames: columns, verticalTimeAxis}, onResetFocus, tableOptions} = this.props
```
Comment (ischolten) [inline ui/src/dashboards/components/TableOptions.tsx]: this is obviously more than a prettier change but it looks like it should be on separate lines.

File: ui/src/dashboards/components/TableOptions.tsx [hunk 5]
```diff
@@ -101,48 +92,37 @@ export class TableOptions extends PureComponent<Props, {}> {
...
 
     const tableSortByOptions = this.computedColumnNames.map(col => ({
       text: col.displayName || col.internalName,
-      key: col.internalName,
+      key: col.internalName
```
Comment (ischolten) [inline ui/src/dashboards/components/TableOptions.tsx]: looks like a prettier change

File: ui/src/dashboards/components/TableOptions.tsx [hunk 6]
```diff
@@ -101,48 +92,37 @@ export class TableOptions extends PureComponent<Props, {}> {
...
     }))
 
     return (
-      <FancyScrollbar
-        className="display-options--cell y-axis-controls"
-        autoHide={false}
-      >
+      <FancyScrollbar className="display-options--cell y-axis-controls" autoHide={false}>
```
Comment (ischolten) [inline ui/src/dashboards/components/TableOptions.tsx]: looks like a prettier change

File: ui/src/dashboards/constants/index.js [hunk 7]
```diff
@@ -20,6 +22,7 @@ export const NEW_DEFAULT_DASHBOARD_CELL = {
   name: 'Untitled Cell',
   type: 'line',
   queries: [],
+  tableOptions: DEFAULT_TABLE_OPTIONS,
```
Comment (ischolten) [inline ui/src/dashboards/constants/index.js]: yay!

File: ui/src/shared/components/TableGraph.js [hunk 8]
```diff
@@ -29,26 +29,38 @@ class TableGraph extends Component {
...
+    const {data, unzippedData} = timeSeriesToTableGraph(nextProps.data)
 
     const {tableOptions: {sortBy: {internalName}}} = nextProps
     const sortByColumnIndex = _.indexOf(data[0], internalName)
 
     const sortedData = _.sortBy(_.drop(data, 1), sortByColumnIndex)
-    this.setState({data: [data[0], ...sortedData], sortByColumnIndex})
+    this.setState({
+      data: [data[0], ...sortedData],
+      unzippedData,
```
Comment (ischolten) [inline ui/src/shared/components/TableGraph.js]: does this need to be initialized in state?

File: ui/src/shared/components/TableGraph.js [hunk 9]
```diff
@@ -29,26 +29,38 @@ class TableGraph extends Component {
...
+  }
 
   handleHover = (columnIndex, rowIndex) => () => {
     if (this.props.onSetHoverTime) {
       const {data} = this.state
-      this.props.onSetHoverTime(data[rowIndex][0].toString())
+      const hoverTime = this.props.tableOptions.verticalTimeAxis
+        ? data[rowIndex][0]
+        : data[0][columnIndex]
+      this.props.onSetHoverTime(hoverTime.toString())
```
Comment (ischolten) [inline ui/src/shared/components/TableGraph.js]: you could peel onSetHoverTime and tableOptions.verticalTimeAxis off of props

File: ui/src/shared/components/TableGraph.js [hunk 10]
```diff
@@ -139,16 +156,20 @@ class TableGraph extends Component {
   render() {
     const {sortByColumnIndex, hoveredColumnIndex, hoveredRowIndex} = this.state
     const {hoverTime, tableOptions, colors} = this.props
-    const {data} = this.state
+
+    const verticalTimeAxis = _.get(tableOptions, 'verticalTimeAxis', true)
+
+    const data = verticalTimeAxis ? this.state.data : this.state.unzippedData
```
Comment (ischolten) [inline ui/src/shared/components/TableGraph.js]: can peal data and unzipped data off of state

File: ui/src/shared/components/TableGraph.js (hunk 8 above)
Comment (ebb-tide) [reply ui/src/shared/components/TableGraph.js]: good catch- thank you !

File: ui/src/shared/components/TableGraph.js (hunk 10 above)
Comment (ebb-tide) [reply ui/src/shared/components/TableGraph.js]: i didn't do this. :/

File: ui/src/dashboards/components/GraphOptionsTimeAxis.js (hunk 1 above)
Comment (ebb-tide) [reply ui/src/dashboards/components/GraphOptionsTimeAxis.js]: I didn't do this.

**Pull request merged**